views/default/elgg/Ajax.js

```diff
@@ -48,6 +49,13 @@ define(function (require) {
 					var params = {
 						options: orig_options
 					};
+					if (data._deps.length) {
+						require(data._deps, function() {
+							ready.resolve(arguments);
+						});
+					} else {
+						ready.resolve();
+					}
 					if (hook_type) {
```

Comment:
I mean, I guess the hooks can't operate synchronously, but that's an inherent limitation.